docs/repositories-configure/using-submodules.md

```diff
@@ -15,12 +15,10 @@ After we enabled submodules for your organization, do the following:
     -   **Public repositories**, make sure that your Git URL uses the HTTPS protocol
     -   **Private repositories**, make sure that your Git URL uses the SSH protocol.
 
-1.  If you already have repositories on Codacy where you want to use submodules, navigate to the page **Settings** and click the button **Generate New User Key** for each of these repositories.
+1.  If you already have repositories on Codacy where you want to use submodules, add a new [SSH key to your git provider account](#user-account-keys) for each of the repositories. On the repository **Settings** page, you can click the link **Add new user key** to go to your Git provider page where you manage your user account SSH keys.
```

Comment:
If I understand correctly, the second sentence explains how to achieve what is described in the first one. Some suggestions:

* Minor tweaks to make this relationship more explicit.
* I'd split the last bit and reformat it to appear as a subsection of the instruction step, since it's contextual information and not an explicit instruction.

Thoughts?

```suggestion
1.  If you already have repositories on Codacy where you want to use submodules, [add a new SSH key to your git provider account](#user-account-keys) for each repository by navigating to the repository **Settings** page and clicking the link **Add new user key**.

    This takes you to the Git provider page of the repository, where you can manage your user account SSH keys.

```

Reply:
I really like @nicklem's suggestions to make the instruction more structured. :+1: 

On the other hand, I'd prefer to have the information behind the link to `#user-account-keys` directly here, as a Note admonition and slightly more condensed. For example, I believe that the following sentence is enough to explain why users need to add a Codacy SSH key to their account:

> When you're using submodules, Codacy needs to clone additional repositories it may not have access to. To overcome this, you must add a Codacy SSH key to your user account to grant Codacy access to the same repositories as your user.

Reply:
Already handled by the [restructured content](https://github.com/codacy/docs/pull/1765#discussion_r1246841017).